File: r/pragmastat/R/measurement.R
# Measurement represents a value with an associated unit.

#' @export
Measurement <- R6::R6Class(
  "Measurement",
  public = list(
    #' @field value Numeric measurement value
    value = NULL,

    #' @field unit MeasurementUnit associated with the value
    unit = NULL,

    #' @description Create a new Measurement
    #' @param value Numeric value
    #' @param unit MeasurementUnit (defaults to number_unit)
    initialize = function(value, unit = NULL) {
      if (is.null(unit)) {
        unit <- number_unit
      }
      self$value <- value
      self$unit <- unit
    }
  )
)

#' @export
as.numeric.Measurement <- function(x, ...) {
  x$value
}

File: py/pragmastat/measurement.py
from dataclasses import dataclass

from .measurement_unit import NUMBER_UNIT, MeasurementUnit


@dataclass(frozen=True)
class Measurement:
    """A value paired with its measurement unit."""

    value: float
    unit: MeasurementUnit = NUMBER_UNIT

    def __float__(self) -> float:
        return self.value
